fastly/custom_tls_certificate.go

```diff
@@ -0,0 +1,195 @@
+package fastly
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// CustomCertificate represents a custom certificate. Uses common TLSDomain type from BulkCertificate
+type CustomCertificate struct {
+	ID                 string       `jsonapi:"primary,tls_certificate"`
+	CreatedAt          *time.Time   `jsonapi:"attr,created_at,iso8601"`
+	IssuedTo           string       `jsonapi:"attr,issued_to"`
+	Issuer             string       `jsonapi:"attr,issuer"`
+	Name               string       `jsonapi:"attr,name"`
+	NotAfter           *time.Time   `jsonapi:"attr,not_after,iso8601"`
+	NotBefore          *time.Time   `jsonapi:"attr,not_before,iso8601"`
+	Replace            bool         `jsonapi:"attr,replace"`
+	SerialNumber       string       `jsonapi:"attr,serial_number"`
+	SignatureAlgorithm string       `jsonapi:"attr,signature_algorithm"`
+	UpdatedAt          *time.Time   `jsonapi:"attr,updated_at,iso8601"`
+	TLSDomains         []*TLSDomain `jsonapi:"relation,tls_domains,tls_domain"`
```

Comment:
Duplication of type again, in case we miss any more can you check the other fields across the PR.

```suggestion
	TLSDomains         []*TLSDomain `jsonapi:"relation,tls_domains"`
```

Reply:
Done